lib/eco/treelexer/lexer.py

```diff
@@ -49,25 +56,50 @@ def match_question(self, pattern, text):
         return self.match(pattern[2:], text)
 
     def match_star(self, pattern, text):
-        if self.match_one(pattern, text):
-            # matched one, continue matching *
-            return self.match(pattern, text[1:])
-        # couldn't match star, remove star regex and continue
-        return self.match(pattern[2:], text)
+        while True:
+            if not self.match(pattern.c, text):
+                break
+        return True
+
+    def match_plus(self, pattern, text):
+        if not self.match(pattern.c, text):
+            # we have to at least match one
+            return False
+        while True:
+            if not self.match(pattern.c, text):
+                break
+        return True
+
+    def match_list(self, pattern, text):
+        for p in pattern:
+            if not self.match(p, text):
+                return False
+        return True
+
+    def match_or(self, pattern, text):
+        tmp = self.pos
+        if self.match(pattern.lhs, text):
+            return True
+        self.pos = tmp # backtrack
+        return self.match(pattern.rhs, text)
 
     def match(self, pattern, text):
+        print("match", pattern, text)
```

Comment:
Stray debugging statement?

Reply:
Probably also removed in the later commits.

Reply:
Probably? ;)